browser/scripts/application.js

```diff
@@ -1026,23 +1030,50 @@ Application.prototype.paste = function(srcDoc, offsetX, offsetY) {
 			var s = graphNode.state
 
 			if (s.input_sids)
-				s.input_sids = mapSlotIds(s.input_sids, uidMap)
+				s.input_sids = mapSlotIds(s.input_sids, localUidMap)
 
 			if (s.output_sids)
-				s.output_sids = mapSlotIds(s.output_sids, uidMap)
+				s.output_sids = mapSlotIds(s.output_sids, localUidMap)
 		}
 
 		graph.conns.map(function(conn) {
-			conn.src_nuid = uidMap[conn.src_nuid]
-			conn.dst_nuid = uidMap[conn.dst_nuid]
+			conn.src_nuid = localUidMap[conn.src_nuid]
+			conn.dst_nuid = localUidMap[conn.dst_nuid]
 			conn.uid = E2.uid()
 		})
 
 		return graph
 	}
 
+	function remapNodeReferences(graph) {
+		graph.nodes.map(function(node) {
+			if (['graph', 'loop', 'array_function'].indexOf(node.plugin) > -1)
```

Comment:
I wonder if this should ask the node instead of relying on its own list. NodeUI has a similar problem already.